agent/lib/kontena/workers/node_info_worker.rb

```diff
@@ -236,6 +253,54 @@ def calculate_container_time(container)
       0
     end
 
+    # @param [Array<Vmstat::Cpu>] prev_cpu_stats
+    # @param [Array<Vmstat::Cpu>] current_cpu_stats
+    # @return [Hash]
+    def calculate_average_cpu(prev_cpu_stats, current_cpu_stats)
+      all = prev_cpu_stats.zip(current_cpu_stats).map do |prev, current|
+        system_ticks = current.system - prev.system
+        user_ticks = current.user - prev.user
+        idle_ticks = current.idle - prev.idle
+
+        total_ticks = system_ticks + user_ticks + idle_ticks
+
+        [system_ticks, user_ticks, idle_ticks].map do |ticks|
+          (ticks / total_ticks.to_f) * 100.0
+        end
+      end
+
+      averages = all.transpose.map do |stats|
+        total = stats.reduce :+
+        total / stats.size.to_f
+      end
+
+      return {
```

Comment:
I know... I actually only used it because it looked so ugly to me starting the line with just the open parens